File: app/graphql/object_types/item_stock_management.rb
# frozen_string_literal: true

module ObjectTypes
  class ItemStockManagement < Types::BaseObject
    field :id, ID, null: false
    field :category, ObjectTypes::Category, null: false
    field :upper_limit, Integer, null: false
    field :item_count, Integer, null: false
    field :created_at, GraphQL::Types::ISO8601DateTime, null: false
    field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
  end
end

File: frontend/src/graphql/fragment/itemStockManagement.ts
import { gql } from "@apollo/client";

export default gql`
  fragment itemStockManagementBase on ItemStockManagement {
    id
    upperLimit
    createdAt
    updatedAt
    itemCount
    category {
      id
      name
      createdAt
      updatedAt
    }
  }
`;